Can't run test suite on macos - envpool not installable, causes ImportError

Greetings,

According to https://tianshou.readthedocs.io/en/latest/contributing.html, I wanted to be able to run unit tests after doing a dev install. Here is what happens on macos:

```
________________________________________________________________________________________________________________ ERROR collecting test/discrete/test_a2c_with_il.py ________________________________________________________________________________________________________________
ImportError while importing test module '/Users/kai7rng/oss/tianshou/test/discrete/test_a2c_with_il.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/discrete/test_a2c_with_il.py:5: in <module>
    import envpool
E   ModuleNotFoundError: No module named 'envpool'
__________________________________________________________________________________________________________________ ERROR collecting test/modelbased/test_psrl.py ___________________________________________________________________________________________________________________
ImportError while importing test module '/Users/kai7rng/oss/tianshou/test/modelbased/test_psrl.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/modelbased/test_psrl.py:5: in <module>
    import envpool
E   ModuleNotFoundError: No module named 'envpool'
```

Maybe tests can have skip statement + import could happen inside individual tests?

--- a/test/continuous/test_sac_with_il.py
+++ b/test/continuous/test_sac_with_il.py
@@ -1,11 +1,5 @@
 import argparse
 import os
-import sys
-
-try:
-    import envpool
-except ImportError:
-    envpool = None
 
 import numpy as np
 import pytest
@@ -18,6 +12,11 @@ from tianshou.trainer import offpolicy_trainer
 from tianshou.utils import TensorboardLogger
 from tianshou.utils.net.common import Net
 from tianshou.utils.net.continuous import Actor, ActorProb, Critic
+
+try:
+    import envpool
+except ImportError:
+    envpool = None
 
 
 def get_args():
@@ -57,8 +56,9 @@ def get_args():
     return args
 
 
-@pytest.mark.skipif(sys.platform != "linux", reason="envpool only support linux now")
+@pytest.mark.skipif(envpool is None, reason="EnvPool doesn't support this platform")
 def test_sac_with_il(args=get_args()):
+    # if you want to use python vector env, please refer to other test scripts
     train_envs = env = envpool.make_gym(
         args.task, num_envs=args.training_num, seed=args.seed
     )

--- a/test/discrete/test_a2c_with_il.py
+++ b/test/discrete/test_a2c_with_il.py
@@ -2,9 +2,9 @@ import argparse
 import os
 import pprint
 
-import envpool
 import gym
 import numpy as np
+import pytest
 import torch
 from torch.utils.tensorboard import SummaryWriter
 
@@ -14,6 +14,11 @@ from tianshou.trainer import offpolicy_trainer, onpolicy_trainer
 from tianshou.utils import TensorboardLogger
 from tianshou.utils.net.common import ActorCritic, Net
 from tianshou.utils.net.discrete import Actor, Critic
+
+try:
+    import envpool
+except ImportError:
+    envpool = None
 
 
 def get_args():
@@ -52,7 +57,9 @@ def get_args():
     return args
 
 
+@pytest.mark.skipif(envpool is None, reason="EnvPool doesn't support this platform")
 def test_a2c_with_il(args=get_args()):
+    # if you want to use python vector env, please refer to other test scripts
     train_envs = env = envpool.make_gym(
         args.task, num_envs=args.training_num, seed=args.seed
     )

--- a/test/modelbased/test_psrl.py
+++ b/test/modelbased/test_psrl.py
@@ -2,8 +2,8 @@ import argparse
 import os
 import pprint
 
-import envpool
 import numpy as np
+import pytest
 import torch
 from torch.utils.tensorboard import SummaryWriter
 
@@ -11,6 +11,11 @@ from tianshou.data import Collector, VectorReplayBuffer
 from tianshou.policy import PSRLPolicy
 from tianshou.trainer import onpolicy_trainer
 from tianshou.utils import LazyLogger, TensorboardLogger, WandbLogger
+
+try:
+    import envpool
+except ImportError:
+    envpool = None
 
 
 def get_args():
@@ -40,7 +45,9 @@ def get_args():
     return parser.parse_known_args()[0]
 
 
+@pytest.mark.skipif(envpool is None, reason="EnvPool doesn't support this platform")
 def test_psrl(args=get_args()):
+    # if you want to use python vector env, please refer to other test scripts
     train_envs = env = envpool.make_gym(
         args.task, num_envs=args.training_num, seed=args.seed
     )

